### ISSUE TITLE
Nested WritableNestedModelSerializer's fail to associate model tables correctly if in reverse alphabetic order

### ISSUE THREAD

**Issue by johnthagen** [beda-software/drf-writable-nested#64]
# Minimal Example

Consider this minimal example:

```py
from django.db import models
from drf_writable_nested import WritableNestedModelSerializer
from rest_framework.serializers import ModelSerializer


class House(models.Model):
    pass


class Cat(models.Model):
    house = models.ForeignKey(House, related_name='cats', on_delete=models.CASCADE, null=True)


class CatPoint(models.Model):
    cat = models.ForeignKey(Cat, related_name='location', on_delete=models.CASCADE, null=True)
    x = models.FloatField()
    y = models.FloatField()


class CatPointSerializer(ModelSerializer):
    class Meta:
        model = CatPoint
        fields = '__all__'


class CatSerializer(WritableNestedModelSerializer):
    location = CatPointSerializer(many=True)

    class Meta:
        model = Cat
        fields = '__all__'


class HouseSerializer(WritableNestedModelSerializer):
    cats = CatSerializer(many=True)

    class Meta:
        model = House
        fields = '__all__'


data = {
    "cats": [
        {
            "location": [
                {
                    "x": 1.0,
                    "y": 2.0
                }
            ]
        }
    ]
}

serializer = HouseSerializer(data=data)
a = serializer.is_valid()
serializer.save()
```

Results in the following traceback:

```
Unhandled exception in thread started by <_pydev_bundle.pydev_monkey._NewThreadStartupWithTrace object at 0x10723b940>
Traceback (most recent call last):
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/backends/sqlite3/base.py", line 296, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.OperationalError: no such table: main.polls_cat__old

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Applications/PyCharm.app/Contents/helpers/pydev/_pydev_bundle/pydev_monkey.py", line 632, in __call__
    return self.original_func(*self.args, **self.kwargs)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/utils/autoreload.py", line 225, in wrapper
    fn(*args, **kwargs)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/core/management/commands/runserver.py", line 117, in inner_run
    self.check(display_num_errors=True)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/core/management/base.py", line 379, in check
    include_deployment_checks=include_deployment_checks,
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/core/management/base.py", line 366, in _run_checks
    return checks.run_checks(**kwargs)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/core/checks/registry.py", line 71, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/core/checks/urls.py", line 13, in check_url_config
    return check_resolver(resolver)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/core/checks/urls.py", line 23, in check_resolver
    return check_method()
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/urls/resolvers.py", line 396, in check
    for pattern in self.url_patterns:
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/utils/functional.py", line 37, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/urls/resolvers.py", line 533, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/utils/functional.py", line 37, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/urls/resolvers.py", line 526, in urlconf_module
    return import_module(self.urlconf_name)
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/user/PycharmProjects/django-test/proj/proj/urls.py", line 20, in <module>
    path('', include('polls.urls')),
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/urls/conf.py", line 34, in include
    urlconf_module = import_module(urlconf_module)
  File "/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/user/PycharmProjects/django-test/proj/polls/urls.py", line 57, in <module>
    serializer.save()
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/drf_writable_nested/mixins.py", line 220, in save
    return super(BaseNestedModelSerializer, self).save(**kwargs)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/rest_framework/serializers.py", line 214, in save
    self.instance = self.create(validated_data)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/drf_writable_nested/mixins.py", line 248, in create
    self.update_or_create_reverse_relations(instance, reverse_relations)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/drf_writable_nested/mixins.py", line 175, in update_or_create_reverse_relations
    related_instance = serializer.save(**save_kwargs)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/drf_writable_nested/mixins.py", line 220, in save
    return super(BaseNestedModelSerializer, self).save(**kwargs)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/rest_framework/serializers.py", line 214, in save
    self.instance = self.create(validated_data)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/drf_writable_nested/mixins.py", line 248, in create
    self.update_or_create_reverse_relations(instance, reverse_relations)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/drf_writable_nested/mixins.py", line 175, in update_or_create_reverse_relations
    related_instance = serializer.save(**save_kwargs)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/rest_framework/serializers.py", line 214, in save
    self.instance = self.create(validated_data)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/rest_framework/serializers.py", line 940, in create
    instance = ModelClass._default_manager.create(**validated_data)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/models/query.py", line 413, in create
    obj.save(force_insert=True, using=self.db)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/models/base.py", line 718, in save
    force_update=force_update, update_fields=update_fields)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/models/base.py", line 748, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/models/base.py", line 831, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/models/base.py", line 869, in _do_insert
    using=using, raw=raw)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/models/query.py", line 1136, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/models/sql/compiler.py", line 1289, in execute_sql
    cursor.execute(sql, params)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/backends/utils.py", line 100, in execute
    return super().execute(sql, params)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/backends/utils.py", line 68, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/backends/utils.py", line 77, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/backends/utils.py", line 85, in _execute
    return self.cursor.execute(sql, params)
  File "/Users/user/.local/share/virtualenvs/proj/lib/python3.7/site-packages/django/db/backends/sqlite3/base.py", line 296, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.OperationalError: no such table: main.polls_cat__old
```

Note the suspicious `polls_cat__old`. This seems to be triggered because how models are created alphabetically:

```bash
$ ./manage.py makemigrations
Migrations for 'polls':
  polls/migrations/0001_initial.py
    - Create model Cat
    - Create model CatPoint
    - Create model House
    - Add field house to cat
```

Note how because `Cat` has a `ForeignKey` to `House`, its final creation is deferred and updated until after `House` is created after `Cat`.

If the code is refactored so that `House` -> `AnimalHouse`, so that it is first alphabetically:

```py
from django.db import models
from drf_writable_nested import WritableNestedModelSerializer
from rest_framework.serializers import ModelSerializer


class AnimalHouse(models.Model):
    pass


class Cat(models.Model):
    house = models.ForeignKey(AnimalHouse, related_name='cats', on_delete=models.CASCADE, null=True)


class CatPoint(models.Model):
    cat = models.ForeignKey(Cat, related_name='location', on_delete=models.CASCADE, null=True)
    x = models.FloatField()
    y = models.FloatField()


class CatPointSerializer(ModelSerializer):
    class Meta:
        model = CatPoint
        fields = '__all__'


class CatSerializer(WritableNestedModelSerializer):
    location = CatPointSerializer(many=True)

    class Meta:
        model = Cat
        fields = '__all__'


class HouseSerializer(WritableNestedModelSerializer):
    cats = CatSerializer(many=True)

    class Meta:
        model = AnimalHouse
        fields = '__all__'


data = {
    "cats": [
        {
            "location": [
                {
                    "x": 1.0,
                    "y": 2.0
                }
            ]
        }
    ]
}

serializer = HouseSerializer(data=data)
a = serializer.is_valid()
serializer.save()
```

Note the output difference in `makemigrations`:

```bash
$ ./manage.py makemigrations
Migrations for 'polls':
  polls/migrations/0001_initial.py
    - Create model AnimalHouse
    - Create model Cat
    - Create model CatPoint
```

Now the same input `data` passes.

## Environment

The exact versions used:

```bash
$ pip freeze
Django==2.1.4
django-rest-framework==0.1.0
djangorestframework==3.9.0
drf-writable-nested==0.5.1
pytz==2018.7
```

Database settings:

```py
DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}
```

# The Problem

Is this a problem in `drf-writable-nested`, DRF itself, something else? Is`WritableNestedModelSerializer` somehow getting a reference to the temporary / old table?

**Comment by johnthagen** [comment]
This might actually just be a Django bug:

- [SQLite 3.26 breaks database migration ForeignKey constraint, leaving <table_name>__old in db schema](https://code.djangoproject.com/ticket/29182)

In Python 3.7.1, when I run:

```
>>> import sqlite3
>>> sqlite3.sqlite_version
'3.26.0'
```

**Comment by johnthagen** [comment]
Confirmed this is fixed in [Django 2.1.5](https://docs.djangoproject.com/en/2.1/releases/2.1.5/#django-2-1-5-release-notes).
